ADR/propagate-pull-secrets.md

```diff
@@ -0,0 +1,50 @@
+# 0. Support deploying images from Private Quay.io repositories
+
+Date: 2023-03-16
+
+## Status
+
+Provisional
+
+## Context
+
+* As a user, I would expect StoneSoup to generate a private Quay.io image repository if my source code repository is  private.
+* As a user, I would expect StoneSoup to gracefully deploy images that were pushed to private Quay.io repositories in every relevant environment.
+
+
+## Decision
+
+### Overview
+
+This proposal builds on the capabilties of the https://github.com/redhat-appstudio/image-controller which sets up a Quay.io image repository per `Component` and 'downloads' a robot account token into the user's namespace for builds to be able to push to it.
+
+#### 1. Scenario: When a new Environment is defined
+
+1. The image-controller copies the pull(actually, push)`Secret` from the user's namespace into the target cluster's namespace for every Component in the workspace/namespace.
+2. The image-controller links the copied `secret` to the `deployer` `serviceAccount`. If the `deployer` `serviceAccount` is absent, the `Secret` 
+    would be linked to the `default` `serviceAccount` to account for distributions of Kubernetes which do not have a `deployer` `serviceAccount`.
+3. At a heartbeat of 24 hours, the `Secret` in the target cluster's namespace is reconciled.
+
+#### 2. Scenario: When a new Component is created
+
+1. As part of the existing 'daily chores' of the image-controller, a new Quay.io image repository and robot account token is made available in the 
```

Comment:
Can you define `daily chores`? Do they run on a cron or does this just describe the normal functionality of the image-controller?